pop-api/src/error.rs

```diff
@@ -0,0 +1,255 @@
+use crate::assets::use_cases::fungibles::{convert_to_fungibles_error, FungiblesError};
+use ink::env::chain_extension::FromStatusCode;
+use scale::{Decode, Encode};
+use PopApiError::*;
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Encode, Decode)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+#[repr(u8)]
+pub enum PopApiError {
```

Comment:
Contract size is a little large, presumably due to the amount and size of the error enums.

One idea is to have a structure similar to the following:
```
enum PopApiError {
   DispatchError {
       index: u8
       nested_index: u8
    },
    ModuleError { index: u8, error: u8 },
    OtherError { ... }
}

// these will only be used if the contract developer wants them. Otherwise, they can offload to a frontend or ignore.
// This means Rust can optimize out unnecessary errors, and reduce overall size.
impl from<PopApiError> for DispatchError {...}
impl from<PopApiError> for ModuleError {...}
...
```